### ISSUE TITLE
CurrentRefinements with Range widgets will show two items instead of one

### ISSUE THREAD

**Issue by Haroenv** [algolia/instantsearch.js#3388]
**Is your feature request related to a problem? Please describe 🙏**

Using CurrentRefinement with a range widget, you will get two separate refinements listed, but realistically only both can be cleared at the same time, since as soon as you touch a range, it should be cleared on both sides

**Describe the solution you'd like 🤔**

The range widget to clarify to the current refinements that those two ranges should be considered as a single one.

**Describe alternatives you've considered ✨**

Some way where we base current refinements off the widgets, instead of the helper state would help in issues like this and #3346

**Comment by justindantzer** [comment]
@Haroenv 

Not to resurrect an old issue, but did you ever find a solution for this issue?

I spent a few hours searching for this exact thing and at some point pieced together a potential solution from a few different sources. (unfortunately i did not note the sources, so I am unable to give credit. if i find the sources, I will update here)

In the hope that this may help someone else, I put together this example:
https://codesandbox.io/s/custom-current-refinements-ykcx2

In this example, I added a custom `currentRefinements` widget via `connectCurrentRefinements`. (see `src/customCurrentRefinements.js`) The refinement items are altered to separate filters that can contain multiple refinements -- for example, the `refinementList`, which allows multiple selections. However, the `numericMenu` filters stay as one unit and are given an alternate label, showing the range as a single string.

`[X] 1 - 3` instead of the normal `[X] > 1 [X] < 3`

The refinements are being stored as a JSON array in a `data-` attribute on the remove button of the refinement item. When the button is clicked, the handler uses the decoded value of the `data-` attribute to refine the upper and lower range values together.

I have not tried this with a range input specifically, but I think (hope) the `numericMenu` handling can be adapted for range inputs.

This is by no means a fully thought out solution, but may prove useful to start down the right path. I'd love to see someone run with it and come up with a solid reusable solution.
